### PR TITLE
Increse number of extraction opportunities in baseline.

### LINKED ISSUE #52: Add fake statements on block leaving in extract semantics step in baseline.
Currently baseline suffers from small amount of large extractions opportunities, which got filtered out easily by syntactic filter. Inserting fake statements on end of each block can help with it.

Based on #51

### PR DESCRIPTION
**Pull request by aravij**
Fix #52, based on #55

Adds fake nodes with empty semantic on extraction semantic step of baseline. It increase number of possible extraction opportunities.

### CODE REVIEW

File: veniq/baselines/semi/create_extraction_opportunities.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def create_extraction_opportunities(
     extraction_opportunities: List[ExtractionOpportunity] = []
     for step in range(1, len(statements_semantic) + 1):
         for extraction_opportunity in _ExtractionOpportunityIterator(statements_semantic, step):
-            if extraction_opportunity not in extraction_opportunities:
+            if len(extraction_opportunity) > 0 and extraction_opportunity not in extraction_opportunities:
```
Comment (Vitaly-Protasov) [inline veniq/baselines/semi/create_extraction_opportunities.py:15]: I think it's more common to use only `if extraction_opportunity` instead

File: veniq/baselines/semi/extract_semantic.py [hunk 2]
```diff
@@ -48,6 +48,14 @@ def on_node_entering(self, node: Union[Block, Statement]) -> None:
         else:
             raise ValueError(f"Unexpected node type {type(node)}.")
 
+    def on_node_leaving(self, node: Union[Block, Statement]) -> None:
+        if isinstance(node, Block):
+            self._on_block_leaving(node)
+        elif isinstance(node, Statement):
```
Comment (Vitaly-Protasov) [inline veniq/baselines/semi/extract_semantic.py:54]:
```
elif not isinstance(node, Statement):
    raise ValueError(f"Unexpected node type {type(node)}.")
```

Looks better, doesn't it?

**Review by Vitaly-Protasov — COMMENTED** [review]
.

File: veniq/baselines/semi/create_extraction_opportunities.py (hunk 1 above)
Comment (aravij) [reply veniq/baselines/semi/create_extraction_opportunities.py:15]: Fixed.

File: veniq/baselines/semi/extract_semantic.py (hunk 2 above)
Comment (aravij) [reply veniq/baselines/semi/extract_semantic.py:54]: I leaved branch with `pass` for further addition. If we would need to handle `Statements` on node leaving we would easily replace `pass` with needed method.

**Review by lyriccoder — APPROVED** [review]
ok

**Pull request merged**